### ISSUE TITLE
Style/RedundantCondition: Use double pipes || instead.

### ISSUE THREAD

**Issue by ianhall** [rubocop/rubocop#11577]
When auto-correcting the below statement:
```
lang = SUPPORTED_LANGS.include? preferred_language ? preferred_language : "en"
```

rubocop replaces the line with:
```
lang = SUPPORTED_LANGS.include? preferred_language || "en"
```

This is clearly wrong, because `.include?` returns `true` or `false`, not the value of `preferred_language`.

Instead, rubocop should suggest the following:
```
lang = SUPPORTED_LANGS.include?(preferred_language) ? preferred_language : "en"
```
where the parameter to the `.include?` is wrapped in parentheses.

This change satisfies the `Style/RedundantCondition` cop without breaking other rules and without changing the functionality of the statement.

--------

## Expected behavior

rubocop should suggest / autocorrect a non-breaking change

## Actual behavior

rubocop suggests a breaking change, which, if the autocorrect is applied, will break the application logic.

```
rubocop --debug
For /Users/ian/dev/lovevery/product_catalog: configuration from /Users/ian/dev/lovevery/product_catalog/.rubocop.yml
configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-rails-2.17.4/config/default.yml
configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-rails-2.17.4/config/default.yml
Default configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-1.45.1/config/default.yml
configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-capybara-2.17.0/config/default.yml
```
... many similar lines excluded
```
configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-rspec-2.18.1/config/default.yml
configuration from /Users/ian/.rvm/gems/ruby-2.7.5@product_catalog/gems/rubocop-rspec-2.18.1/config/default.yml
Inheriting configuration from /Users/ian/dev/lovevery/product_catalog/.rubocop-todo.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/ian/.cache/rubocop_cache/d96b2bd1f63c5a516ab7c9b4d98904e2713a2475/d0167bc68559d895266a8b1332feb7079d4bac3f/26faaa983184b1d90b812acae9c8180a58ea2577
Loading cache from /Users/ian/.cache/rubocop_cache/d96b2bd1f63c5a516ab7c9b4d98904e2713a2475/d0167bc68559d895266a8b1332feb7079d4bac3f/37c4c74a223f1ce14b063128445643ec54abdfd8
.

Offenses:

app/services/stripe_service/api_client.rb:119:80: C: [Correctable] Style/RedundantCondition: Use double pipes || instead.
      lang = SUPPORTED_LANGS.include? preferred_language ? preferred_language : "en"
                                                        ^^^^^^^^^^^^^^^^^^^^^^

655 files inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.7440480000223033 seconds
```

## Steps to reproduce the problem

Create a file thus:
```
class Test
  SUPPORTED_LANGS = %w[en de fr].freeze
  def rubo(preferred_language)
    SUPPORTED_LANGS.include? preferred_language ? preferred_language : "en"
  end
end
```

with a `.rubocop.yml` file which does not disable `Style/RedundantCondition` and run `rubocop -A`

## RuboCop version

```
$ [bundle exec] rubocop -V
bundle exec rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 2.7.5) [arm64-darwin22]
  - rubocop-rails 2.17.4
  - rubocop-rspec 2.18.1
 ```

**Comment by koic** [comment]
As a short answer this is an expected behavior of `Style/RedundantCondition`.

The reproducible code is expressed in AST as follows:

```ruby
% ruby-parse -e 'SUPPORTED_LANGS.include? preferred_language ? preferred_language : "en"'
(send
  (const nil :SUPPORTED_LANGS) :include?
  (if
    (send nil :preferred_language)
    (send nil :preferred_language)
    (str "en")))
```

So the ternary operator is the argument, and adding the parentheses gives:

```ruby
SUPPORTED_LANGS.include?(preferred_language ? preferred_language : "en")
```

`SUPPORTED_LANGS.include? preferred_language || "en"` is compatible, `SUPPORTED_LANGS.include?(preferred_language) ? preferred_language : "en"` is incompatible.

Parentheses were used against the original code and would need to be modified to the intended code.

**Comment by ianhall** [comment]
Thank you very much for the explanation @koic - I appreciate it!
